File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        mp = {}
        for ind, ele in enumerate(nums):
            # print(mp.get(target-ele))
            sec = mp.get(target-ele)
            if sec is not None:
                return [sec, ind]
            mp[ele] = ind
        # print(mp)
        return []

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer> mp = new HashMap<>();
        for(int i=0;i<nums.length;i++){
            int ele = mp.getOrDefault(target-nums[i],-1);
            if(ele != -1){
                return new int[]{i,ele};
            }
            mp.put(nums[i], i);
        }

        return new int[0];
    }
}